src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -158,6 +171,49 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
     /** Periods which have just been added. */
     const addedPeriods: ITSPeriodObject[] = [];
 
+    for (const period of periods) {
+      ["audio" as const, "video" as const].forEach((ttype: ITrackType) => {
+        const forType = period.adaptations[ttype];
+        const periodHasAdaptationForType = forType !== undefined && forType.length > 0;
+        if (!periodHasAdaptationForType) {
+          log.debug(
+            `TS: The period does not have adaptation for ${ttype} there is no track to choose`,
+          );
+          return;
+        }
+        const firstPlayableAdaptation = findFirstPlayableAdaptation(period, ttype);
+        if (
+          forType !== undefined &&
+          forType.every((a) => a.supportStatus.hasSupportedCodec === false)
+        ) {
+          const err = new MediaError(
+            "MANIFEST_INCOMPATIBLE_CODECS_ERROR",
+            "No supported " + ttype + " adaptations",
+            { tracks: undefined },
+          );
+          if (
+            firstPlayableAdaptation === undefined &&
+            ttype === "audio" &&
+            this.onAudioTracksNotPlayable === "continue"
+          ) {
+            // Audio is not playable but video may be playable, let's continue the playback.
+            this.trigger("warning", err);
+          } else if (
+            firstPlayableAdaptation === undefined &&
+            ttype === "video" &&
+            this.onVideoTracksNotPlayable === "continue"
+          ) {
+            // Video is not playable but audio may be playable, let's continue the playback.
+            this.trigger("warning", err);
+          } else if (firstPlayableAdaptation !== undefined) {
+            this.trigger("warning", err);
+          } else {
+            this.trigger("error", err);
```

Comment:
Shouldn't we also just abort the trackstore here? That's what seem to be done for `NO_PLAYABLE_REPRESENTATION`

Reply:
Done